Add style to Polymer version on main page and detail page

kul3r4 opened the pull request:
@sebastianbenz This adds: 

- Style to the main page localhost:8080/polymer. Please note that since I needed to change the css and it only works with absolute URLs, in order to run it locally you need to copy inside the `style` of `sample-home` the added css (which was moved from the template `index.html`)

- Anchor support

- I am trying a different approach from how AMPByExample is at the moment on the main page. Instead of displaying all the content (categories and samples description), I would like to try and experiment showing less in the main page (only the categories) and having an extra page for the category which shows the samples for the category. The reason for this is that I feel like there is too much content in the home page and too much scrolling to do. 
Let me know what you think, this is also the reason I haven't styled the list page yet, just in case you say that is not doable at all.

sebastianbenz commented:
Why doesn't the css work? Would a separate subdomain, e.g. polymer.ampyexample.com help?

kul3r4 commented:
It's explained in the [bug](https://github.com/Polymer/polymer-cli/issues/892), it's not related where the css is hosted, but more how the build checks for css files only inside the polymer folder.

sebastianbenz commented on polymer/index.html:
```diff
@@ -10,11 +10,15 @@
-    <script src="/polymer/node_modules/@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>
-
-    <link rel="import" href="/polymer/src/abe-polymer-app.html">
+    <link rel="stylesheet" type="text/css" href="http://localhost:8080/css/example.css">
```
localhost urls?

kul3r4 replied:
Sorry, that was a leftover

sebastianbenz commented on polymer/src/sample-detail.html:
```diff
@@ -96,15 +98,6 @@
-        hidden$="[[!failure]]"
-        offline="[[offline]]"
-        on-try-reconnect="_tryReconnect"></shop-network-warning>
-
```
why did you remove the offline warning?

kul3r4 replied:
Because I am not adding that component at the moment, for simplicity.

sebastianbenz commented on polymer/src/sample-category-data.html:
```diff
@@ -26,6 +26,11 @@
   <script>
   (function() {
 
+    function sanitizeString(string) {
```
What does this method do?

kul3r4 replied:
It sanitizes strings like this: ""<p>Get started with AMP and learn how to build your first AMP page.</p>\n""

kul3r4 commented:
@sebastianbenz available on [staging](https://amp-by-example-staging.appspot.com/polymer/)

sebastianbenz commented:
I like the new navigation. In the case of the home screen you could then remove the no longer necessary sidebar.

Few things I've noticed looking at the sample:

* Make the whole card clickable and remove the action
* page load feels quite slow (too slow)
* sometimes pages flicker as if they're loaded multiple times
* category pages are un-styled
* scroll position is not reset
* console displays quite a few 404s
* white frame around content

![Uploading Screen Shot 2017-10-04 at 4.25.43 PM.png…]()

kul3r4 commented:
Hi, I don't think the scroll position should be reset.
From [https://medium.com/@owencm/designing-great-uis-for-progressive-web-apps-dd38c1d20f7#.ofgfr058y](https://medium.com/@owencm/designing-great-uis-for-progressive-web-apps-dd38c1d20f7#.ofgfr058y):

> tapping back should bring back to the list page with the same scroll position they were at before they pressed on the item.

kul3r4 commented:
@sebastianbenz PTAL, new version is on [staging](https://amp-by-example-staging.appspot.com/polymer)

kul3r4 commented:
@sebastianbenz PTAL

kul3r4 commented:
@sebastianbenz:
center cards on desktop -> done
make it possible to navigate from detail pages to category and home page -> I added the header on top. For now, the ads section will point to the original one, not the polymer one.
Can you transition between pages, e.g. fade? -> I added a fade effect on the list of cards

kul3r4 commented:
@sebastianbenz ping

sebastianbenz commented:
Is it live on staging?

On Fri, Nov 3, 2017 at 12:55 PM Chiara Chiappini <notifications@github.com>
wrote:

> @sebastianbenz <https://github.com/sebastianbenz> ping
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ampproject/amp-by-example/pull/951#issuecomment-341695652>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAXOOHex0e4FMEUQY4_bSKtGXiFQodC0ks5syw1SgaJpZM4Pqz70>
> .
>

kul3r4 commented:
Yes, and I just noticed that I missed the menu for mobile (it only appears for desktop)

sebastianbenz commented:
Noticed a few issues with the demo:

* does not work in desktop safari 
* cards are not correctly aligned on desktop
* pink card content (after moving back and forth)
* cards stopped working after a few navigations, see console log
* cards fade in, but sample pages don't (can you show a loading indicator while the page is loading? Otherwise there's only a blank page visible)
* few errors on initial page load in console

![screen shot 2017-11-06 at 9 13 36 am](https://user-images.githubusercontent.com/380472/32434007-cd53e7da-c2d4-11e7-8d06-0ac6448cc8c0.png)
![screen shot 2017-11-06 at 9 13 42 am](https://user-images.githubusercontent.com/380472/32434008-cd9d9b6e-c2d4-11e7-93c6-c22bd557bf3d.png)
<img width="1535" alt="screen shot 2017-11-06 at 9 22 26 am" src="https://user-images.githubusercontent.com/380472/32434009-cdd668b8-c2d4-11e7-82b3-62ac6d3e096a.png">
![screen shot 2017-11-06 at 9 14 12 am](https://user-images.githubusercontent.com/380472/32434010-ce11504a-c2d4-11e7-84a7-228374aab1ab.png)
![screen shot 2017-11-06 at 9 14 26 am](https://user-images.githubusercontent.com/380472/32434012-ce516e50-c2d4-11e7-8f1f-0d1ea45644e2.png)

kul3r4 commented:
@sebastianbenz Latest version is on [staging](https://amp-by-example-staging.appspot.com/polymer)
